src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -369,6 +370,15 @@ private ConstantValue GetLazyConstantValue(bool earlyDecodingWellKnownAttributes
             Debug.Assert((GetLazyConstantValue(earlyDecodingWellKnownAttributes) == Microsoft.CodeAnalysis.ConstantValue.Unset) ||
                 (GetLazyConstantValue(earlyDecodingWellKnownAttributes) == value));
 
+            // Check availability of `DecimalConstantAttribute`.
```

Comment:
>// Check availability of `DecimalConstantAttribute`. [](http://example.com/codeflow?start=12&length=52)

It looks like there is a same issue with performing a check when the attribute is explicitly applied. See what ```SourceMemberFieldSymbol.AddSynthesizedAttributes``` is doing in this case.